safe-defaults/dnsmasq-opendns-family-shield.conf

```diff
@@ -0,0 +1,2 @@
+strict-order
+server=208.67.222.123
```

Comment:
I struggled to find documentation on exactly how NetworkManager generates the dnsmasq configuration. Here is what I found in `man 5 NetworkManager.conf`:

```
       dns
           Set the DNS processing mode
...
           dnsmasq: NetworkManager will run dnsmasq as a local caching nameserver, using "Conditional Forwarding" if you are
           connected to a VPN, and then update resolv.conf to point to the local nameserver. It is possible to pass custom
           options to the dnsmasq instance by adding them to files in the "/etc/NetworkManager/dnsmasq.d/" directory. Note that
           when multiple upstream servers are available, dnsmasq will initially contact them in parallel and then use the
           fastest to respond, probing again other servers after some time. This behavior can be modified passing the
           'all-servers' or 'strict-order' options to dnsmasq (see the manual page for more details).
...
           When using dnsmasq and systemd-resolved per-connection added dns servers will always be queried using the device the
           connection has been activated on.
```

So that last paragraph implies, I think (but I may be misunderstanding it!), the behaviour we hope for: the dnsmasq configuration is generated by concatenating all the files in `/etc/NetworkManager/dnsmasq.d/*.conf`, and then appending the DNS servers from the connection itself.

Reply:
I think it is the `--server` mentioned in [DNSMASQ's manual](https://thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html).  dnsmasq can direct the query to the corresponding nameserver by the domain name.  Here does not specify which domain name here, so it means all domain shall be directed to the nameserver `208.67.222.123`.  If this nameserver does not response & timeout, which means it's dead, then goes to next nameserver.